application/shared-kernel/ApiCore/Endpoints/TrackEndpoints.cs

```diff
@@ -0,0 +1,206 @@
+                        Url = new Uri(trackRequestDto.Data.BaseData.Url),
+                        Duration = trackRequestDto.Data.BaseData.Duration,
+                        Timestamp = trackRequestDto.Time,
+                        Id = trackRequestDto.Data.BaseData.Id,
+                        PerfTotal = trackRequestDto.Data.BaseData.PerfTotal,
+                        NetworkConnect = trackRequestDto.Data.BaseData.NetworkConnect,
+                        SentRequest = trackRequestDto.Data.BaseData.SentRequest,
+                        ReceivedResponse = trackRequestDto.Data.BaseData.ReceivedResponse,
+                        DomProcessing = trackRequestDto.Data.BaseData.DomProcessing
+                    };
+
+                    CopyContextTags(telemetry.Context, trackRequestDto);
+                    CopyDictionary(trackRequestDto.Data.BaseData.Properties, telemetry.Properties);
+                    CopyDictionary(trackRequestDto.Data.BaseData.Measurements, telemetry.Metrics);
+
+                    telemetryClient.Track(telemetry);
+                    break;
+                }
+                case "ExceptionData":
+                {
+                    var exceptionDetailsInfos = GetExceptionDetailsInfos(trackRequestDto);
+                    var telemetry = new ExceptionTelemetry(exceptionDetailsInfos,
+                        trackRequestDto.Data.BaseData.SeverityLevel, trackRequestDto.Data.BaseType,
+                        trackRequestDto.Data.BaseData.Properties, new Dictionary<string, double>())
+                    {
+                        SeverityLevel = trackRequestDto.Data.BaseData.SeverityLevel,
+                        Timestamp = trackRequestDto.Time
+                    };
+
+                    CopyContextTags(telemetry.Context, trackRequestDto);
+                    CopyDictionary(trackRequestDto.Data.BaseData.Properties, telemetry.Properties);
+                    CopyDictionary(trackRequestDto.Data.BaseData.Measurements, telemetry.Metrics);
+
+                    telemetryClient.TrackException(telemetry);
+                    break;
+                }
+                case "MetricData":
+                {
+                    var metric = trackRequestDto.Data.BaseData.Metrics[0];
+                    var telemetry = new MetricTelemetry
+                    {
+                        Name = metric.Name,
+                        Sum = metric.Value,
+                        Count = metric.Count,
+                        Timestamp = trackRequestDto.Time
+                    };
+
+                    CopyContextTags(telemetry.Context, trackRequestDto);
+                    CopyDictionary(trackRequestDto.Data.BaseData.Properties, telemetry.Properties);
+
+                    telemetryClient.TrackMetric(telemetry);
+                    break;
+                }
+                case "RemoteDependencyData":
+                    // Ignore remote dependency data
```

Comment:
Should we not log warning here too? And if yes, then just remove this case and handle it by the `default:` case? If not, we should document, why we ignore this.